### ISSUE TITLE
Doesn't compile anymore with v0.6.1 of stm32f30x which got released yesterday

### ISSUE THREAD

**Issue by Caemor** [japaric/stm32f30x-hal#23]
Compiling returns an error since the update of stm32f30x yesterday. 

It seems like a function was changed to a unsafe function.

```
    Checking stm32f30x v0.6.1
    Checking stm32f30x-hal v0.1.2
error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:189:25
    |
189 | /                         w.sadd1()
190 | |                             .bits(addr)
    | |_______________________________________^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:189:25
    |
189 | /                         w.sadd1()
190 | |                             .bits(addr)
191 | |                             .rd_wrn()
192 | |                             .clear_bit()
193 | |                             .nbytes()
194 | |                             .bits(bytes.len() as u8)
    | |____________________________________________________^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:207:49
    |
207 |                           self.i2c.txdr.write(|w| w.txdata().bits(*byte));
    |                                                   ^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:237:25
    |
237 | /                         w.sadd1()
238 | |                             .bits(addr)
    | |_______________________________________^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:237:25
    |
237 | /                         w.sadd1()
238 | |                             .bits(addr)
239 | |                             .rd_wrn()
240 | |                             .clear_bit()
241 | |                             .nbytes()
242 | |                             .bits(bytes.len() as u8)
    | |____________________________________________________^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:255:49
    |
255 |                           self.i2c.txdr.write(|w| w.txdata().bits(*byte));
    |                                                   ^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:263:25
    |
263 | /                         w.sadd1()
264 | |                             .bits(addr)
    | |_______________________________________^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error[E0133]: call to unsafe function requires unsafe function or block
   --> /home/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/stm32f30x-hal-0.1.2/src/i2c.rs:263:25
    |
263 | /                         w.sadd1()
264 | |                             .bits(addr)
265 | |                             .rd_wrn()
266 | |                             .set_bit()
267 | |                             .nbytes()
268 | |                             .bits(buffer.len() as u8)
    | |_____________________________________________________^ call to unsafe function
...
291 | / hal! {
292 | |     I2C1: (i2c1, i2c1en, i2c1rst),
293 | |     I2C2: (i2c2, i2c2en, i2c2rst),
294 | | }
    | |_- in this macro invocation

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0133`.
error: Could not compile `stm32f30x-hal`.
```

**Comment by japaric** [comment]
Apparently svd2rust v0.12.0, the version used to generate the version v0.6.0 of stm32f30x, had a bug where methods that ought to be `unsafe` (due to the SVD file not containing enough information) were not and this crate was making use of it. v0.12.1 of svd2rust, which was used to generate the version v0.6.1of stm32f30x, fixed that bug and broke this crate.

I have patched the stm32f30x SVD file to make those I2C methods safe again (there was nothing unsafe about them, but the SVD didn't provide enough information to svd2rust telling it that was the case), released v0.6.2 of stm32f30x and yanked v0.6.1. 

TL;DR if you update your Cargo.lock file this should fix itself
